sdk/nodejs/tests/runtime/props.spec.ts

```diff
@@ -95,18 +143,111 @@ describe("runtime", () => {
         }));
         it("marshals secrets correctly", asyncTest(async () => {
             runtime._setTestModeEnabled(true);
+
             const inputs: Inputs = {
                 "secret1": secret(1),
                 "secret2": secret(undefined),
             };
+
             // Serialize and then deserialize all the properties, checking that they round-trip as expected.
-            const transfer = gstruct.Struct.fromJavaScript(
+            runtime._setFeatureSupport("secrets", true);
+            let transfer = gstruct.Struct.fromJavaScript(
                 await runtime.serializeProperties("test", inputs));
-            const result = runtime.deserializeProperties(transfer);
+            let result = runtime.deserializeProperties(transfer);
+            assert.ok(runtime.isRpcSecret(result.secret1));
+            assert.ok(runtime.isRpcSecret(result.secret2));
+            assert.strictEqual(runtime.unwrapRpcSecret(result.secret1), 1);
+            assert.strictEqual(runtime.unwrapRpcSecret(result.secret2), null);
+
+            // Serialize and then deserialize all the properties, checking that they round-trip as expected.
+            runtime._setFeatureSupport("secrets", false);
+            transfer = gstruct.Struct.fromJavaScript(
+                await runtime.serializeProperties("test", inputs));
+            result = runtime.deserializeProperties(transfer);
             assert.strictEqual(result.secret1, 1);
             assert.strictEqual(result.secret2, undefined);
+
             runtime._setTestModeEnabled(false);
         }));
+        it("marshals resource references correctly during preview", asyncTest(async () => {
+            runtime._setTestModeEnabled(false);
```

Comment:
Would it make sense to have `_reset` call `_setTestModeEnabled(false)` (or perhaps more correctly reset it to `undefined` by directly poking into `options`)? Then, we can throw `_reset` into a `beforeEach`. Technically, this line is a no-op for this test since we'll have a monitor below, right? It took me a couple minutes to try to reason about what this is really doing and by getting rid of it, that might help people who look at this later down the road (instead of trying to reason through all the test mode toggling we do throughout this code)

Reply:
I was not aware of `beforeEach` (and neither, it seems, were the original authors of these tests)--that sounds like a good idea. I think that `_reset` will already disable test mode, as it clears the `options`.

Reply:
I'll do this as a follow-up. I'm already touching a bit more code than I'd like.